presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1116,6 +1119,13 @@ else if (frame.getType() == GROUPS) {
                 }
             }
 
+            if (node.isIgnoreNulls()) {
```

Comment:
@auden-woolfson : I generally preferred this check to be in the block for window since the validation is specific for it. But if we want to keep this check here, then we should also add node.getWindow().isPresent() to this condition.

Reply:
@aditi-pandit,

For the first option of bringing this logic into the `node.getWindow().isPresent()` conditional block, this would require restructuring the entire function. Since we need the handle, all of the logic surrounding `argumentTypesBuilder` (and a few other lines of code) would need to be put at the top of the function.

I tried the second option of adding the window is present condition to the ignore nulls condition and this is causing tests to fail. `node.getWindow()` is not present when `node.isIgnoreNulls()`. I think the window is only present when the functions to the right of the `OVER` clause in the SQL query are being visited.

The tests are passing, I believe the current implementation is sound.

Reply:
@auden-woolfson : I am fine with the current changes.